[게시판 회원 기능] 철시 미션 제출합니다.

pkch93 opened the pull request:
현구님 안녕하세요~! 이번에 현구님께 리뷰를 받게 된 철시입니다!

현재 테스트가 미비한 상태입니다. 테스트 위주로 개발하려고 노력했지만 제 생각대로 되지 않아서 일단은 먼저 프로덕션 코드라도 짠 다음 피드백을 받고자 이렇게 리뷰요청보냅니다,,

테스트는 회원이 등록되었다면 먼저 post요청으로 값을 등록하고 이후에 redirect 되는 경로를 체크하는 방식으로 고민해 보았는데 등록자체가 안되더라고요... repository를 주입하고 해봐도 테스트가 깨져서 일단 테스트 부분은 어떻게 해야할지 고민중입니다.

그리고 이번에 미션을 하면서 JPA를 사용하고 있잖아요.
이 과정에서 처음에는 제가 Entity 클래스를 비즈니스 로직을 가지는 도메인 객체라고 생각하고 그렇게 짜는게 맞겠다 생각하여 현재 짠 구조는 그에 맞춰 구현하였습니다.

그런데 문득 Entity가 결국에는 DB와 맞닿아 있는 부분이라는 생각이 들었어요. 그래서 Entity를 도메인 객체로 사용하는 게 맞는걸까? 라는 의문을 가졌는데 현구님 생각은 어떤지 궁금합니다 ㅎㅎ

많이 부족한 코드입니다. 많은 지적 부탁드립니다!

seonghun127 committed:
setting: mission complete

seonghun127 committed:
docs: README.md

1. todo 목록 정리

seonghun127 committed:
feat: build.gradle

1. devtools 의존성 추가

seonghun127 committed:
feat: ArticleController Class

1. Article 도메인 객체 구현
2. Test 추가
3. ArticleRepository 저장기능 구현

seonghun127 committed:
feat: layout

1. build.gradle 의존성 추가
2. 공통 레이아웃 추출 및 적용

seonghun127 committed:
feat: ArticleDto Class

1. Create 용도의 ArticleDto 구현
2. ArticleDto를 사용하도록 Controller와 Repository 변경

seonghun127 committed:
refactor: Article class

1. ArticleDto를 사용하기 위한 생성자 추가
2. id 자동 증가를 위한 static field 추가

seonghun127 committed:
feat: ArticleController class

1. 조회 기능 구현

seonghun127 committed:
style: css 및 templates 파일 코드 포메팅

seonghun127 committed:
fix: app.css

1. 파일 오류 수정

seonghun127 committed:
test: ArticleRepository class

1. 조회 및 전체조회 테스트

seonghun127 committed:
fix: layout

1. thymeleaf-layout 버전에 따른 오류 수정

seonghun127 committed:
fix: ArticleController class

1. createArticle 리다이렉팅 주소 수정

seonghun127 committed:
feat: ArticleController class

1. 게시물 수정 기능 구현
2. ArticleDto.Update 생성
3. ArticleRepository에 update 기능 추가
4. update를 위해 Article 도메인 객체에 생성자 추가

seonghun127 committed:
feat: article-edit.html

1. 생성 화면과 수정 화면에 따른 form 경로 및 메서드 조정
2. article.html에 하드코딩된 url 변경

seonghun127 committed:
feat: ArticleController class

1. 삭제 기능 구현
2. ArticleRepository delete 구현

seonghun127 committed:
feat: article.html

1. delete 버튼 구현

seonghun127 committed:
refactor: layout

1. header 공통부분 추출

seonghun127 committed:
fix: ArticleRepository class

1. 수정시 새 게시물이 추가되는 오류 수정

seonghun127 committed:
style: 전체 코드 포매팅

pkch93 committed:
fix: css 적용 오류 해결

pkch93 committed:
refactor: article.html

기본 템플릿으로 주어지는 initValue 및 viewer 삭제

pkch93 committed:
refactor: ArticleControllerTests

1. 각 테스트에서 반복적으로 나타나는 하드코딩 문자열 상수화

pkch93 committed:
refactor: ArticleRepositoryTest

1. 각 테스트에서 반복적으로 나타나는 매직넘버 상수화 (1)

pkch93 committed:
refactor: ArticleControllerTests

1. get 요청 중복 부분 메서드화하여 리펙토링

codeanddonuts committed:
[게시글 생성/조회/수정/삭제] 도넛 미션 제출합니다. (#11)

* init

* feat: Article POJO

* feat: Article repository

* feat: router

* fix: article delete

* fix: put method

* test: controller tests

* test: repo tests

* feat: live reloading and auto refresh support

* fix : layout

* doc: README.md

* refactor: constructor added

* refactor: validations refactored

* refactor: ModelAndView to RedirectView

* fix: ArticleRepo test

* refactor: moved application properties settings from main method to a separate file

* refactor: tests

* refactor: numbering method

* refactor: no sending http method from the servre explicitly for branching

* refactor: uri routing

codeanddonuts committed:
feat: mysql connection settings

codeanddonuts committed:
refactor: merged ArticleDTO to Article class.

codeanddonuts committed:
feat: JPA applied.

codeanddonuts committed:
Merge branch 'codeanddonuts' into level2

codeanddonuts committed:
fix: tests

codeanddonuts committed:
pray: please...

codeanddonuts committed:
test: removed repo test

codeanddonuts committed:
tests: router

codeanddonuts committed:
doc: README.md

codeanddonuts committed:
feat: user feature controller

codeanddonuts committed:
feat: User domain

codeanddonuts committed:
test: User

codeanddonuts committed:
feat: login page

codeanddonuts committed:
test: login

codeanddonuts committed:
feat: user list

codeanddonuts committed:
feat: User repo

codeanddonuts committed:
feat: user list view

codeanddonuts committed:
feat: signup and validation

codeanddonuts committed:
feat: login and session management

codeanddonuts committed:
fix: session and cookie

codeanddonuts committed:
feat: front-end profile differentiation to login status

codeanddonuts committed:
feat: personal user info edit

codeanddonuts committed:
fix: user info edit

codeanddonuts committed:
feat: user exit

pkch93 committed:
feat: login

login error 메세지 도출 구현

pkch93 committed:
docs: README.md

리팩토링 고려할 부분 정리

pkch93 committed:
refactor: repository package

domain 내부에 있던 Repository 관련 인터페이스 파일을 repository 패키지로
분리

pkch93 committed:
settings: test 환경 db 설정

pkch93 committed:
test: 메서드 이름 변경

pkch93 committed:
feat: UserDto class

클라이언트에게서 값을 주고 받도록 하기 위한 UserDto 클래스 구현

pkch93 committed:
feat: UserDto class

Login 관련 Dto 추가

pkch93 committed:
feat: AuthInterceptor

인증 관련하여 인터셉터 구현

pkch93 committed:
refactor: UserDto class

각 컬럼들이 모두 같은 제약조건을 가지는 데
중복으로 제약 조건을 줘야하는 상황보다 하나로 합치는 것이 낫다고 판단.
상황별로 나눈 Dto를 하나로 합침

pkch93 committed:
refactor: User class

User 엔티티 클래스를 도메인 객체에 맞게 리펙토링

pkch93 committed:
refactor: UserController class

login 관련 Exception 처리 및 로직 리팩토링

pkch93 committed:
settings: application.properties

session 사용시 url 뒤에 jsessionid가 붙지 않도록 하기위해
cookie 설정

pkch93 committed:
test: UserController class

login 정상 흐름 테스트 추가

pkch93 committed:
refactor: UserController class

UserController로 들어오는 요청을 /auth/** 로 변경

pkch93 committed:
feat: User, UserDto class

수정시 결국은 JPA가 식별자를 보고 판단하므로 id를 넘기는 작업이
필요하다고 판단.

때문에 UserDto에 id를 추가하고 User에 id를 인자로 갖는 생성자 추가

pkch93 committed:
feat: ModelTranslator class

도메인 객체와 User 모델을 합쳐주는 Translator 구현

pkch93 committed:
refactor: UserController class

회원가입 로직 리팩토링

pkch93 committed:
refactor: UserController class

mypage, mypageedit get 요청 처리 로직 리팩토링

pkch93 committed:
refactor: AuthController, UserController class

UserController에 있는 로그인 / 로그아웃 인증 로직 AuthController로 분리

pkch93 committed:
refactor: UserController class

수정, 삭제 로직 리팩토링

pkch93 committed:
fix: AuthController class

경로 오류 발견, @RequestMapping 추가로 해결

pkch93 committed:
refactor: User class

User의 모든 값을 가지는 생성자 private 변경

pkch93 committed:
settings: application.properties

MySQL 연결 설정

pkch93 committed:
refactor: auth.html

로그인, 로그아웃 경로 재설정

pkch93 committed:
refactor: login.html, header.html

- 로그인 POST 요청 경로 재설정
- 기존에 session의 key값으로 로그인 여부를 판단한 것을 email 여부로 판단

pkch93 committed:
refactor: templates

url 변경에 따른 링크 경로 수정

pkch93 committed:
fix: UserDto class

email Pattern 추가

pkch93 committed:
fix: mypage.html

수정 페이지로 넘어가지 못하는 문제 해결
href 경로를 잘못 설정한 문제

pkch93 committed:
fix: application.properties

mysql 한글 깨짐 현상 해결

pkch93 committed:
fix: signup.html

상위 헤더의 우아한테크코스 아이콘 이동경로 수정

pkch93 committed:
fix: UserController class

User update시 세션에 반영안되는 문제 수정

pkch93 committed:
fix: UserController, AuthController class

에러메세지 어노테이션 변경, 그에따른 templates 수정

pkch93 committed:
fix: NeedAuthInterceptor class

로그인 된 상태에서 login 페이지, signup 페이지 접근시 리다이렉팅이
안되는 문제 해결

pkch93 committed:
fix: WebConfig class

인터셉터 설정 오류 변경, /users의 POST 방식으로 회원가입시 login페이지로
리다이렉팅 되는 오류 해결

pkch93 committed:
fix: UserTranslator class

User 수정시 수정이 제대로 안되는 문제 해결

pkch93 committed:
refactor: Article class

Article 엔티티 클래스를 Domain 객체에 맞게 변환

pkch93 committed:
feat: ArticleDto class

Article 관련 값 전송을 위한 ArticleDto 클래스 구현

pkch93 committed:
feat: ArticleTranslator class

Article 엔티티 객체와 ArticleDto를 합쳐주는 ArticleTranslator 구현

pkch93 committed:
refactor: ArticleController class

pkch93 committed:
docs: README.md

pkch93 committed:
style: 전체 소스코드 포메팅 및 최적화

pkch93 committed:
Revert "[게시글 생성/조회/수정/삭제]  철시 미션 제출합니다. (#22)"

This reverts commit 2fe7a043da1325f914dcf0e5b1f8b8a6e122de00.

pkch93 committed:
Merge branch 'dev' into pkch93

kang-hyungu commented:
1. 회원 등록 테스트를 실행해보니 작성한 uri 경로가 없어요. 인수 테스트 코드들에 uri가 맞는지 다시 확인해보세요
2.  저는 jpa를 사용하면 entity와 domain을 분리해서 관리하지 않습니다. 하나로 묶어서 보고, 실제 구현한 domain 객체를 보면 db 테이블과 1:1 매핑을 하지 않고 되도록 객체 설계에 맞게 구성해요. 예를 들면 User 클래스에 Email이라는 클래스를 인스턴스 변수로 가질 수 있는데 실제 rdb는 그런 구조를 한 테이블로 표현하지 못하죠. 애초에 어노테이션 용도도 메타 데이터(meta data) 목적이기도 하죠.
그래서 저는 entity 어노테이션을 사용한 객체를 domain 객체로 사용해도 문제 없다고 생각합니다.

kang-hyungu reviewed (changes requested):
안녕하세요 철시님 리뷰어 강현구입니다.
dto에서 도메인 객체 사이에 translator를 두고 객체 변환을 관리하는 방법도 괜찮네요 :+1: 
객체 변환에 사용하는 컨버터 패턴이라는 것도 있으니 참고해보세요
https://github.com/iluwatar/java-design-patterns/tree/master/converter
질문 주신 내용은 코멘트 남겼으니 확인해보시고, 피드백 남긴 부분도 보시고 코드에 반영하시고 재요청 주세요!

pkch93 committed:
style: build.gradle 포메팅

pkch93 committed:
refactor: Interceptor classes

인터셉터 구현체 클래스 web 패키지에서 config로 이동

pkch93 committed:
refactor: ModelTranslator implements classes

ModelTranslator를 구현하고 있는 클래스들을 Spring 빈 팩토리가 관리하지
않도록 만듦

Translator가 스프링 빈으로 등록하지 않음에 따라 Controller에서 Translator
를 생성하여 사용하도록 변경

pkch93 committed:
test: User class

User 관련 테스트 추가

pkch93 committed:
test: Article class

Article 관련 테스트 추가

pkch93 committed:
settings: logback-access 설정

pkch93 committed:
refactor: UserController class

System.out.println 제거

pkch93 committed:
fix: WebConfig class

비로그인 상태에서 로그아웃 시 에러 수정

pkch93 committed:
test: AuthController, UserController class

인증 관련 요청 및 유저 관련 요청 인수 테스트

pkch93 committed:
test: ArticleController class

글쓰기 관련 로직 인수테스트

pkch93 committed:
refactor: ArticleRepository, UserRepository

CrudRepository -> JpaRepository

pkch93 committed:
refactor: UserDto class

email 필드에 필요없는 어노테이션 @Email 제거

pkch93 committed:
feat: UserService class

컨트롤러에서 사용하는 로직을 서비스로 추상화
그과정에서 User를 email로도 구별할 수 있도록 User class에 메서드 추가

pkch93 committed:
refactor: UserController class

UserService 적용

pkch93 committed:
feat: AuthService class

AuthController의 비즈니스 로직을 담은 AuthService 구현

pkch93 committed:
refactor: NotMatchPasswordException class

현재 인증을 이메일과 비밀번호를 받아서 하고 있으므로 이에 맞게
NotMatchAuthenticatedException으로 변경

pkch93 committed:
refactor: AuthController class

AuthService 적용

pkch93 committed:
style: 전체 소스코드 포메팅

pobiconan reviewed (approved):
피드백 반영하느라 수고했어요. 💯 
다음 단계에도 다른 리뷰어와 즐거운 경험 하기를 바래요.
